src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -35,8 +35,9 @@ function ShallowWaterSolutionFields(grid, bcs, prognostic_names)
     return NamedTuple{prognostic_names[1:3]}((u, v, h))
 end
 
-mutable struct ShallowWaterModel{G, A<:AbstractArchitecture, T, GR, V, U, R, F, E, B, Q, C, K, TS, FR} <: AbstractModel{TS}
+mutable struct ShallowWaterModel{G, A<:AbstractArchitecture, NL, T, GR, V, U, R, F, E, B, Q, C, K, TS, FR} <: AbstractModel{TS}
                           grid :: G         # Grid of physical points on which `Model` is solved
+                vertical_model :: NL        # Either single-layer or multi-layer
```

Comment:
maybe `layer_formulation`? (we use the word "formulation" to refer to different ways of writing the equations, ie `VectorInvariantFormulation` and `ConservativeFormulation`... maybe we can reuse that word here.)

Reply:
I do like the economy and literalness of simply writing the number of layers though

Reply:
I also like the simplicity of a type field or constructor kwarg called `number_of_layers` instead of the tad more esoteric `vertical_model`.

Reply:
No inclinations here, so happy to revert back to `number_of_layers`.